tests/functional/helpers.js

```diff
@@ -6,11 +6,18 @@ export const checkItemsInList = async (itemsExpected, listLabel) => {
   }
 };
 
-export const checkListItemComplete = async (listItemLabel) => {
-  await expect(await $(listItemLabel).$(`.ons-summary__item-title-icon.ons-summary__item-title-icon--check`).isExisting()).toBe(true);
+export const summaryItemComplete = async (summaryItemLabel) => {
```

Comment:
Do we need to duplicate these methods with complete/incomplete, could just pass the bool in? Then we'd only need two?

Reply:
Or even one if we passed the element?

Reply:
Done, I did think about that but there would be some logic needed for the selector so I decided to leave it as two as it seemed neater to me. Happy to make the change to one though if other people feel different

Reply:
You trade readability for universality so there are pros and cons for both approaches

Reply:
```
export const itemComplete = async (itemLabel, listType, status) => {
  let selector;
  if (listType === "Summary") {
    selector = `.ons-summary__item-title-icon.ons-summary__item-title-icon--check`;
  } else if (listType === "List") {
    selector = `.ons-list__prefix.ons-list__prefix--icon-check`;
  }
  await expect(await $(itemLabel).$(selector).isExisting()).toBe(status);
};
```
This is the sort of thing we would be looking at just to give you an idea

Reply:
Why don't we just pass in the actual element instead of listType? So you don't need the "ifs"?

Reply:
Because the selectors are quite long I felt like passing them in every time wasn't the best for readability. I could do that by setting them to variables at the top of the test files I spose but I feel like doing all this to just remove one helper statement is a bit overkill

Reply:
I mean, the function definition describes well what the parameters are, if we get rid of the "if" logic and not set the hardcoded vars we end up with only 1 or 2 lines of code for this reusable function. I prefer that, after some consideration.

Reply:
Yeah I understand that, my thoughts were just that it would be simple in this file but you'd be just passing the problem down to the test file. It would mean that we would need to pass in `.ons-summary__item-title-icon.ons-summary__item-title-icon--check` or `.ons-list__prefix.ons-list__prefix--icon-check`, the selector we already pass in and the status every time we used this function which seems less readable to me. We could set up consts in the test file to help with this but all in all it comes down to whether we call a different function (list vs summary) or we pass in an extra param (list vs summary) which would be a long one, in that way its much of a muchness when you are calling it but is less readable. It just feels like to me we are trying to make this function too generic

Reply:
Might be easier to chat about it tomorrow in tech session

Reply:
If we can identify same pattern in different functions and use one instead of two or four that do essentially the same task that is a good thing. But if it is less readable to people keep it separate, down to the devs.

Reply:
I'd usually say more generic is better, but agree that it needs to be readable. The reason I commented here in the first place as the fact we had multiple methods meant we were actually using the wrong one initially (when we thought the test might be flaky), and that's always a risk when there's a bunch of methods doing pretty much the same thing, so more maintainable and clearer sometimes when there is only one.

Reply:
Yeah agreed. Happy to have the conversation and explain a bit more in tech session, I think that will make this discussion easier